examples/crossplane-xr-multiversion/demo.sh

```
@@ -229,39 +442,68 @@ pe "kubectl get xwidgets.v1.example.org from-v2 -n ${NS} -o yaml | sed -n '/^spe
+elif [[ "${DEMO_MODE}" == gitops ]]; then
+  note "GitOps retarget: generate Kyverno policies, apply, admission strips pins and sets xrd-api-version=v2."
+  pe "convctl generate kyverno --xrd 03-promote-v2/xrd.yaml --from v1 --to v2"
+  pe "kubectl apply -f gitops/policies/from-v1-to-v2.yaml"
+else
+  note "Retarget every XR onto xwidgets-v2.example.org and drop the pinned revision:"
+  pe "cat patches/retarget-v2.json"
+  pe 'for xr in $(kubectl get xwidgets.example.org -n '"${NS}"' -o name); do kubectl patch "$xr" -n '"${NS}"' --type=json --patch-file patches/retarget-v2.json; done'
+fi
+if [[ "${DEMO_MODE}" == gitops ]]; then
+  note "Kyverno 1.18.1 never runs MutatingPolicy mutateExisting (kyverno#16255). Admission needs a write."
+  pe 'for xr in $(kubectl get xwidgets.example.org -n '"${NS}"' -o name); do kubectl annotate "$xr" -n '"${NS}"' demo.example.org/retarget="$(date +%s)" --overwrite; done'
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)